src/main.rs

```diff
@@ -582,14 +591,26 @@ fn run_shell(vm: &VirtualMachine, scope: Scope) -> PyResult<()> {
     let stdout = io::stdout();
     let mut stdout = stdout.lock();
 
-    print!("{}", get_prompt(vm, "ps1"));
-    stdout.flush().expect("flush failed");
+    let mut print_prompt = || {
```

Comment:
This does not need to be `mut`. You might also define a normal function like `fn print_prompt() {` here.